src/skmultiflow/prototype/rslvq.py

```diff
@@ -0,0 +1,395 @@
+                    "length of prototypes per class"
+                    " does not fit the number of classes"
+                    "classes=%d"
+                    "length=%d" % (nb_classes, nb_ppc.size))
+
+        # initialize prototypes
+        if self.initial_prototypes is None:
+            if self.initial_fit:
+                self.w_ = np.empty([np.sum(nb_ppc), nb_features],
+                                   dtype=np.double)
+                self.c_w_ = np.empty([nb_ppc.sum()], dtype=self.classes_.dtype)
+            pos = 0
+            for actClassIdx in range(len(self.classes_)):
+                actClass = self.classes_[actClassIdx]
+                nb_prot = nb_ppc[actClassIdx]  # nb_ppc: prototypes per class
+                if (self.protos_initialized[actClassIdx] == 0 and
+                        actClass in unique_labels(train_lab)):
+                    mean = np.mean(
+                        train_set[train_lab == actClass, :], 0)
+
+                    if self.prototypes_per_class == 1:
+                        # If only one prototype we init it to mean
+                        self.w_[pos:pos + nb_prot] = mean
+                    else:
+                        # else we add some random noise to distribute them
+                        self.w_[pos:pos + nb_prot] = mean + (
+                            random_state.rand(nb_prot, nb_features) * 2 - 1)
+
+                    if math.isnan(self.w_[pos, 0]):
+                        raise ValueError('Prototype on position {} for class\
+                                         {} is NaN.'
+                                         .format(pos, actClass))
+                    else:
+                        self.protos_initialized[actClassIdx] = 1
+
+                    self.c_w_[pos:pos + nb_prot] = actClass
+                pos += nb_prot
+        else:
+            x = validation.check_array(self.initial_prototypes)
+            self.w_ = x[:, :-1]
+            self.c_w_ = x[:, -1]
+            if self.w_.shape != (np.sum(nb_ppc), nb_features):
+                raise ValueError("the initial prototypes have wrong shape\n"
+                                 "found=(%d,%d)\n"
+                                 "expected=(%d,%d)" % (
+                                     self.w_.shape[0], self.w_.shape[1],
+                                     nb_ppc.sum(), nb_features))
+            if set(self.c_w_) != set(self.classes_):
+                raise ValueError(
+                    "prototype labels and test data classes do not match\n"
+                    "classes={}\n"
+                    "prototype labels={}\n".format(self.classes_, self.c_w_))
+        if self.initial_fit:
+            if self.gradient_descent == 'adadelta':
+                self.squared_mean_gradient = np.zeros_like(self.w_)
+                self.squared_mean_step = np.zeros_like(self.w_)
+            self.initial_fit = False
+
+        return train_set, train_lab
+
+    def fit(self, X, y, classes=None):
```

Comment:
The `fit` and `partial_fit` methods in `ClassifierMixin` also takes the parameter `sample_weight`. Given that this method does not use it, you can write in the docstring:

```
sample_weight: Not used.
```

It is important to keep this parameter to avoid potential errors due to a mismatch in the signature. For example, if this method is used as base learner in an ensemble.